### PR TITLE
fix(parameters): Avoid multiple providers fetch calls

### PR DESCRIPTION
**Pull request by sBouzols**
Move states where needed
Add some useRef to control effect processing
Rename AdvancedParameterButton -> SubgroupParametersButton

### CODE REVIEW

File: src/components/dialogs/parameters/parameters.js [hunk 1]
```diff
@@ -333,6 +309,27 @@ export const useParametersBackend = (
         ]
     );
 
+    useEffect(() => {
+        if (user !== null) {
+            backendFetchProviders()
+                .then((providers) => {
+                    // we can consider the provider get from back will be also used as
```
Comment (laurgarn) [inline src/components/dialogs/parameters/parameters.js:316]: get -> gotten

**Review by laurgarn — APPROVED** [review]
Code : OK
Test : OK
console : no worse

File: src/components/dialogs/parameters/load-flow-parameters.js [hunk 2]
```diff
@@ -253,12 +253,9 @@ const AdvancedParameterButton = ({ showOpenIcon, label, callback }) => {
...
 };
 
-const AdvancedLoadFlowParameters = ({
-    lfParams,
-    commitLFParameter,
-    showAdvancedLfParams,
-    setShowAdvancedLfParams,
-}) => {
+const AdvancedLoadFlowParameters = ({ lfParams, commitLFParameter }) => {
+    const [showAdvancedLfParams, setShowAdvancedLfParams] = useState(false);
```
Comment (dbraquart) [inline src/components/dialogs/parameters/load-flow-parameters.js:257]: moving these states into inner components has a little behaviour impact:
when we expand "advanced parameters" and/or "specific parameters", and then we visit others tabs from the Parameters dialog, and go back to LF parameters,  then "advanced parameters" and/or "specific parameters" are not expanded anymore.

This is a small acceptable change (to me), but I wanted to report it.

File: src/components/dialogs/parameters/parameters.js [hunk 3]
```diff
@@ -265,14 +248,7 @@ export const useParametersBackend = (
...
             });
-    }, [
-        type,
-        backendFetchDefaultProvider,
-        providers,
-        provider,
-        updateProvider,
-        snackError,
-    ]);
+    }, [type, backendFetchDefaultProvider, updateProvider, snackError]);
```
Comment (dbraquart) [inline src/components/dialogs/parameters/parameters.js:251]: yes, removing 'providers' is  a good point (seems to avoid multiple effects)

**Review by dbraquart — APPROVED** [review]

File: src/components/dialogs/parameters/parameters.js (hunk 1 above)
Comment (sBouzols) [reply src/components/dialogs/parameters/parameters.js]: fixed even if not my text

File: src/components/dialogs/parameters/load-flow-parameters.js (hunk 2 above)
Comment (sBouzols) [reply src/components/dialogs/parameters/load-flow-parameters.js:257]: You are right ! It's a regression.
Fixed, tell we if it's ok now ?

File: src/components/dialogs/parameters/parameters.js (hunk 3 above)
Comment (sBouzols) [reply src/components/dialogs/parameters/parameters.js:251]: Did you see why this effect was even calling `resetProvider` sometimes...
this was random, depending on two fetch performance... amazing...

Comment (dbraquart) [reply src/components/dialogs/parameters/parameters.js:251]: non did not see that

File: src/components/dialogs/parameters/load-flow-parameters.js (hunk 2 above)
Comment (dbraquart) [reply src/components/dialogs/parameters/load-flow-parameters.js:257]: good!

**Review by dbraquart — APPROVED** [review]

**Pull request merged**